libpoolprotocols/PoolManager.cpp

```diff
@@ -247,10 +247,11 @@ void PoolManager::stop()
     DEV_BUILD_LOG_PROGRAMFLOW(cnote, "PoolManager::stop() end");
 }
 
-void PoolManager::addConnection(URI& conn)
+void PoolManager::addConnection(URI& _conn)
 {
-    Guard l(m_activeConnectionMutex);
```

Comment:
Why removed ?  Preventing another API call modifying connections the same time

Reply:
I pushed a test commit nevertheless all mutexes are not necessary as the API lives in the same thread. There is only one thread for all which is not a miner

Reply:
In other words ethminer threads are (1 + device number)

Reply:
All actions from any number of API connections are executed serialized by io_service.
There is no concurrency to worry about. There's to worry about asynchrony. What can happen is that a call (eg. Setactiveconnection) does not complete it's whole cycle before another call kicks in